### PR TITLE
Fix--collection-activity-chart

### LINKED ISSUE #7427: Chart showing wrong numbers on Dolphins?
https://canary.kodadot.xyz/ahp/collection/13/activity?sale=true

image

fix should go to the `main` tho

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #7427

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

## Screenshot 📸

- [x] My fix has changed UI

image

### <samp>🤖 Generated by Copilot at 37acb56</samp>

This pull request improves the collection activity component by fixing bugs, optimizing performance, and enhancing the chart display. It uses a utility function to calculate optimal bin sizes for the chart, formats buy event values according to the chain decimals, and avoids redundant calls to `displayMoreEvents`. It also ensures that the last event is shown in the `Events.vue` component.

### <samp>🤖 Generated by Copilot at 37acb56</samp>

> _`displayValue` shines_
> _Chart bins adjust to data_
> _Winter collection_

### CODE REVIEW

File: components/collection/activity/ActivityChart.vue [hunk 1]
```diff
@@ -43,16 +46,26 @@ const listEvents = computed(() => {
...
 
   const binnedBuyEvents = buyBins.map(({ timestamp, value }) => [
     new Date(timestamp),
-    displayValue(value),
+    displayValue(value, decimals.value),
   ])
   const binnedListEvents = listBins.map(({ timestamp, value }) => [
     new Date(timestamp),
-    displayValue(value),
+    displayValue(value, 10),
```
Comment (Jarsen136) [inline components/collection/activity/ActivityChart.vue:68]: Why hard coding?
```suggestion
    displayValue(value, decimals.value),
```

Comment (daiagi) [reply components/collection/activity/ActivityChart.vue:64]: absolutly no reason. thanks for catching it

Comment (daiagi) [reply components/collection/activity/ActivityChart.vue]: fixed

**Review by Jarsen136 — COMMENTED** [review]
otherwise lgtm

File: components/collection/activity/events/Events.vue [hunk 2]
```diff
@@ -27,11 +27,14 @@
 
     <template #rows="{ variant }">
       <EventRow
-        v-for="(event, i) in displayedEvents"
+        v-for="(event, i) in displayedEvents.slice(0, displayedEvents.length)"
```
Comment (Jarsen136) [inline components/collection/activity/events/Events.vue:30]: Is there anything different compared to the previous one? Copy an array?
```suggestion
        v-for="(event, i) in displayedEvents
```

Comment (daiagi) [reply components/collection/activity/events/Events.vue:30]: >Copy an array?

not taking the last array element

the table of events was not loading all the way
point is to render it 1 before the last event row

suggestions on how to do it better are welcome
before:
```
<EventRow />
<EventRow />
...
<div ref="sentinal"/>
```

After:
```
<EventRow />
<EventRow />
...
<EventRow />
<div ref="sentinal"/>
<EventRow/>

Comment (Jarsen136) [reply components/collection/activity/events/Events.vue:30]: > > Is there anything different compared to the previous one? Copy an array?
> 
> not taking the last array element
> 

However, it does not do as you expected.  

image

Comment (daiagi) [reply components/collection/activity/events/Events.vue:30]: of course, what an embarrassing mistake
fixed.
Thank you

**Review by Jarsen136 — APPROVED** [review]

**Comment by yangwao** [comment]
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 4.25 USD/DOT ~ 7.059 $DOT
🧗 [1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8](https://polkadot.subscan.io/account/1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)
🔗 [0x2184963355801cc939a2fe56c9a1fa86c942356b68646bfd58b2219a17417323](https://polkadot.subscan.io/extrinsic/0x2184963355801cc939a2fe56c9a1fa86c942356b68646bfd58b2219a17417323)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**